src/prost/build.rs

```diff
@@ -99,7 +99,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         // Eq + Hash are for plan nodes to do common sub-plan detection.
         // The requirement is from Source node -> SourceCatalog -> WatermarkDesc -> expr
         .type_attribute("catalog.WatermarkDesc", "#[derive(Eq, Hash)]")
-        .type_attribute("catalog.StreamSourceInfo", "#[derive(Eq, Hash)]")
+        .type_attribute("catalog.StreamSourceInfo", "#[derive(Eq)]")
```

Comment:
please respect the comment above

Reply:
Similar to the dependency path the comment shows, `StreamSourceInfo` is required for path `Source node -> SourceCatalog -> StreamSourceInfo `

common sub-plan detection required the eq comparison of source node. Changing the implementation of the `Hash` to hashing part of `StreamSourceInfo` will not cause incorrect eq comparison. This has been confirmed with @chenzl25 

> Can the Eq be deleted?

No. See reasons above